ast/src/main/scala/jawn/ast/JValue.scala

```diff
@@ -143,7 +145,7 @@ object JNum { self =>
 
   final def hybridEq(x: Long, y: Double): Boolean = {
     val z = x.toDouble
-    y == z && z.toLong == x
+    y == z && y.toLong == x
```

Comment:
So, the idea here is that there are large `Long` values which can't be exactly represented as a `Double`. Here's an example:

```
scala> val x: Long = (1L << 55) + 1L
x: Long = 36028797018963969

scala> val z = x.toDouble
z: Double = 3.6028797018963968E16

scala> z.toLong
res0: Long = 36028797018963968

scala> z.toLong == x
res1: Boolean = false
```

Normally when you compare a `Long` and a `Double` in Scala, Scala will convert the `Long` into a `Double` first. This means that a `Long` will be equal to the "closest" `Double`, even if they aren't exactly the same. By doing `z.toLong == x` we are ensuring that the `Double` does convert back to the original `Long`. If this fails it means the `Long` was too bad (as in the REPL example) and so the comparison should fail.

Does that make sense? It seems like I should have added a comment explaining what is going on there.